### DISCUSSION TITLE
Order by / sort

### DISCUSSION THREAD

**Discussion by davetapley** [msiemens/tinydb#510] [General]
I'm a bit surprised there is no order by / sort functionality. Am I missing something?

For my use case I'm inserting documents with a `{'_date': '2023-02-09T03:28:13.625084'` field,
so I'd like to be able to sort on that and then `.get(` the most recent document matching the query.

**Comment by msiemens** [comment]
Hey @davetapley, you're not missing anything! TinyDB indeed does not have a sort/order by feature.

> so I'd like to be able to sort on that and then .get( the most recent document matching the query.

Regarding `get()`, there are no guarantees about which document is returned if multiple documents match the query. Thus, using `get()` together with _order by_ would no be guaranteed to work anyway, but I agree that it might be interesting to sort the results that `search()` returns.

However, I fear introducing sort/order by would lead to a lot of complexity, particularly around corner cases like sorting by a nested field, sorting by multiple fields etc. For this reason I'd probably be inclined to not implement order by/sorting at all and leaving this to the user by sorting the result of `search()` themselves.

Does this make sense to you?
